src/main/msp/msp_serial.c

```diff
@@ -561,7 +571,8 @@ int mspSerialPush(uint8_t cmd, uint8_t *data, int datalen, mspDirection_e direct
 
     for (int portIndex = 0; portIndex < MAX_MSP_PORT_COUNT; portIndex++) {
         mspPort_t * const mspPort = &mspPorts[portIndex];
-        if (!mspPort->port) {
+
+        if (!mspPort->port || !mspPort->isDisplayPort) {
```

Comment:
With this check in place, would it make sense to pull out the 'Avoid zombie VCP port' (added in #1422) below - only sending displayPort on ports that are explicitly enabled for it should do the trick.